app/controllers/api_controller/service_dialogs.rb

```diff
@@ -14,5 +14,44 @@ def show_service_dialogs
       end
       show_generic(:service_dialogs)
     end
+
+    def refresh_dialog_fields_resource_service_dialogs(type, id = nil, data = nil)
+      raise BadRequestError, "Must specify an id for Reconfiguring a #{type} resource" unless id
+
+      api_action(type, id) do |klass|
+        service_dialog = resource_search(id, type, klass)
+        api_log_info("Refreshing Dialog Fields for #{service_dialog_ident(service_dialog)}")
+
+        refresh_dialog_fields_service_dialog(service_dialog, data)
+      end
+    end
+
+    private
+
+    def service_dialog_ident(service_dialog)
+      "Service Dialog id:#{service_dialog.id} label:'#{service_dialog.label}'"
+    end
+
+    def refresh_dialog_fields_service_dialog(service_dialog, data)
+      data ||= {}
+      dialog_fields = Hash(data["dialog_fields"])
```

Comment:
Tried that, but instance variable set tests seems to only work on controllers, and checking value after the completion was useless that the value was already updated by the action.